trinity/main.py

```diff
@@ -134,14 +119,19 @@ def main():
         # TODO: actually use args.sync_mode (--sync-mode)
         sync_mode = SYNC_LIGHT
 
-    chain_config = ChainConfig.from_parser_args(chain_identifier, args)
+    chain_config = ChainConfig.from_parser_args(
+        chain_identifier,
+        args,
+    )
 
     # if console command, run the trinity CLI
     if args.subcommand == 'console':
         use_ipython = not args.vanilla_shell
         debug = args.log_level.upper() == 'DEBUG'
 
-        chain = chain_obj(chain_config, sync_mode)
+        chain_class = get_chain_protocol_class(chain_config, sync_mode)
+        chaindb = BaseChainDB(LevelDB(chain_config.database_dir))
+        chain = chain_class(chaindb)
```

Comment:
Looks like a reasonable solution to the earlier `sync_mode` question 👍

Reply:
Yeah, it cleaned itself up on it's own once I separated initialization of the data-dir from initialization of the database.